pocs/utils/logger.py

```diff
@@ -28,10 +28,16 @@ class StrFormatLogRecord(logging.LogRecord):
     def getMessage(self):
         msg = str(self.msg)
         if self.args:
-            try:
-                msg = msg % self.args
-            except (TypeError, ValueError):
-                msg = msg.format(*self.args)
+            if '{' in msg:
```

Comment:
Were there cases that weren't being covered?

Reply:
Yes. IIRC, it was when some non-primitive type (dict, I think) was passed in as the args (e.g. for {!r}), but as a result the % operator didn't complain.

Reply:
Got it. I never use the `{!r}` syntax and don't feel like I've seen the error but it's certainly more robuse.